pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -468,8 +476,8 @@ func (cr *PerconaXtraDBCluster) CheckNSetDefaults(serverVersion *ServerVersion)
 		}
 	}
 
-	if cr.Spec.UpdateStrategy == SmartUpdateStatefulSetStrategyType && !cr.Spec.ProxySQL.Enabled {
-		return false, fmt.Errorf("ProxySQL should be enabled if SmartUpdate set")
+	if cr.Spec.UpdateStrategy == SmartUpdateStatefulSetStrategyType && !cr.Spec.ProxySQL.Enabled && !cr.Spec.HAProxy.Enabled {
+		return false, fmt.Errorf("ProxySQL or HAProxy should be enabled if SmartUpdate set")
```

Comment:
It's gonna be panic if SmartUpdate is switched on (nil pointer dereference of cr.Spec.ProxySQL or cr.Spec.HAProxy)`

And earlier we have `c := cr.Spec`. Please make this statement consistent with the rest of the method (s/cr.Spec.ProxySQL/c.ProxySQL etc.)

Reply:
@dAdAbird I don't understand where is the panic gonna be ?

Reply:
@AlyHKafoury if cr.Spec.ProxySQL or cr.Spec.HAProxy wasn't set in the CR it is gonna be nil. And when you try to access a filed of the nil object (e.g. cr.Spec.ProxySQL.Enabled) panic happens. See https://play.golang.org/p/51GOu16yoQJ

Everywhere else we're checking if ProxySQL and HAProxy isn't nil (e.g. https://github.com/percona/percona-xtradb-cluster-operator/pull/443/files#diff-ec2d08f17b8825b0a0aaf77aa563359aR407)